### PR TITLE
build: Add help to main Makefile

### PR DESCRIPTION
**Pull request by jmrodri**
Add `help` target to main Makefile. 

```
$ make help
Usage: make <OPTIONS> ... <TARGETS>

Available targets are:

art                  generate artwork
version              show the current version
program              show the project name
path                 create working paths
deps                 install system and golang dependencies
run                  run mgmt
bindata              generate go files from non-go sources
lang                 generate languages
build                build a fresh mgmt binary
clean                clean things up
test                 run tests
format               format yaml and go code
docs                 generate docs
upload               upload sources
copr                 build in copr
deb                  build debian package
build_container      builds the container
clean_container      removes the container
help                 Show this help screen
```

### CODE REVIEW

**Comment by jmrodri** [comment]
make-help-mgmt

File: Makefile [hunk 1]
```diff
@@ -88,48 +88,48 @@ art/mgmt_logo_white_wide.png: art/mgmt_logo_white_wide.svg
 all: docs $(PROGRAM).static
 
 # show the current version
-version:
+version: ## show the current version
 	@echo $(VERSION)
 
-program:
+program: ## show the project name
```
Comment (purpleidea) [inline Makefile]: should this be s/project/program/ ?

File: Makefile [hunk 2]
```diff
@@ -88,48 +88,48 @@ art/mgmt_logo_white_wide.png: art/mgmt_logo_white_wide.svg
...
-bindata:
+bindata: ## generate go files from non-go sources
 	@echo "Generating: bindata..."
 	$(MAKE) --quiet -C bindata
 
 generate:
 	go generate
 
-lang:
+lang: ## generate languages
```
Comment (purpleidea) [inline Makefile]: can you reword this one please?
it generates the lexer/parser for the lang frontend

File: Makefile [hunk 3]
```diff
@@ -179,9 +179,9 @@ gofmt:
 yamlfmt:
 	find . -maxdepth 3 -type f -name '*.yaml' -not -path './old/*' -not -path './tmp/*' -not -path './omv.yaml' -exec ruby -e "require 'yaml'; x=YAML.load_file('{}').to_yaml.each_line.map(&:rstrip).join(10.chr)+10.chr; File.open('{}', 'w').write x" \;
 
-format: gofmt yamlfmt
+format: gofmt yamlfmt ## format yaml and go code
```
Comment (purpleidea) [inline Makefile]: s/go/golang/ please

File: Makefile [hunk 4]
```diff
@@ -314,30 +314,39 @@ upload-rpms: rpmbuild/RPMS/ rpmbuild/RPMS/SHA256SUMS rpmbuild/RPMS/SHA256SUMS.as
...
 	docker cp mgmt-build:/root/gopath/src/github.com/purpleidea/mgmt/mgmt .
 	docker build -t purpleidea/mgmt -f docker/Dockerfile.static .
 	docker rm mgmt-build || true
 
-clean_container:
+clean_container: ## removes the container
 	docker rmi purpleidea/mgmt-build
 	docker rmi purpleidea/mgmt
 
+help: ## Show this help screen
```
Comment (purpleidea) [inline Makefile]: nit: this is the only help message which is capitalized, want to make it uniform and lower case it please?

File: Makefile [hunk 5]
```diff
@@ -314,30 +314,39 @@ upload-rpms: rpmbuild/RPMS/ rpmbuild/RPMS/SHA256SUMS rpmbuild/RPMS/SHA256SUMS.as
...
+	@echo 'Usage: make <OPTIONS> ... <TARGETS>'
+	@echo ''
+	@echo 'Available targets are:'
+	@echo ''
+	@grep -E '^[ a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | \
+	    awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}'
```
Comment (purpleidea) [inline Makefile]: indent with tabs, no spaces please.

**Comment by jmrodri** [comment]
@purpleidea I made the changes based on your comments. I also amended the commit message, I hope that fixes the tests. Also, I didn't add every task to the help since some of them seem more internal than something the user would interact with directly. Let me know what you think, do we want to document all of the targets or just the relevant ones. Thanks for looking at my PR.

**Comment by purpleidea** [comment]
one bug: https://travis-ci.org/purpleidea/mgmt/jobs/363566485#L1462

capitalize "add" in commit

File: Makefile [hunk 6]
```diff
@@ -88,48 +88,48 @@ art/mgmt_logo_white_wide.png: art/mgmt_logo_white_wide.svg
...
 
-lang:
+lang: ## generates the lexer/parser for the language frontend
 	@# recursively run make in child dir named lang
 	@echo "Generating: lang..."
 	$(MAKE) --quiet -C lang
 
 # build a `mgmt` binary for current host os/arch
-$(PROGRAM): build/mgmt-${GOHOSTOS}-${GOHOSTARCH}
+$(PROGRAM): build/mgmt-${GOHOSTOS}-${GOHOSTARCH} ## build a mgmt binary for current host os/arch
```
Comment (purpleidea) [inline Makefile]: build a mgmt binary for current host os/arch  ->  build an mgmt binary for current host os/arch

please :) a -> an

**Comment by purpleidea** [comment]
Thanks for the great PR!

File: Makefile (hunk 6 above)
Comment (jmrodri) [reply Makefile]: Fixed. How is `mgmt` pronounced? I thought it was 'management' or is it 'em jee em tee'?

**Comment by jmrodri** [comment]
@purpleidea should be done now, thanks for your patience.

**Comment by purpleidea** [comment]
(Note: the GitHub Pull request topic name has nothing to do with the commit message -- it's a common source of confusion. You need to fix with --amend)

**Comment by purpleidea** [comment]
\0/ awesome! MERGED :)

Thanks for your patience, and figuring out the quirks of all our test cases!
This is pretty cool magic. Let us know if you're interested in more hacking.

**Pull request merged**